pkg/recommendation/resource_events.go

```diff
@@ -0,0 +1,64 @@
+package recommendation
+
+import (
+	"github.com/kubeshop/botkube/pkg/config"
+	"github.com/kubeshop/botkube/pkg/events"
+	"github.com/kubeshop/botkube/pkg/ptr"
+)
+
+const (
+	podsResourceName    = "v1/pods"
+	ingressResourceName = "networking.k8s.io/v1/ingresses"
+)
+
+// ResourceEventsForConfig returns the resource event map for a given source recommendations config.
+func ResourceEventsForConfig(recCfg config.Recommendations) map[string]config.EventType {
+	resNames := make(map[string]config.EventType)
+
+	if ptr.IsTrue(recCfg.Ingress.TLSSecretValid) || ptr.IsTrue(recCfg.Ingress.BackendServiceValid) {
+		resNames[ingressResourceName] = config.CreateEvent
+	}
+
+	if ptr.IsTrue(recCfg.Pod.NoLatestImageTag) || ptr.IsTrue(recCfg.Pod.LabelsSet) {
+		resNames[podsResourceName] = config.CreateEvent
+	}
+
+	return resNames
+}
+
+// ShouldIgnoreEvent returns true if user doesn't listen to events for a given resource, apart from enabled recommendations.
+func ShouldIgnoreEvent(recCfg config.Recommendations, sources map[string]config.Sources, sourceBindings []string, event events.Event) bool {
```

Comment:
Reading this here makes sense, but over in the controller it's a bit confusing why an event is ignored.

Can we rename this to something that better explains the intent?

Reply:
I believe the function says exactly what it does (tells if an event should be ignored), and the comment explains when a given event could be ignored. Also, it is in the `recommendation` package so it tells that it is related somehow to recommendations. Honestly, I don't have any other ideas which are not too long. Do you have any? 🙂 

---

Alternatively, WDYT about such change:

```go
	if recommendation.ShouldIgnoreEvent(recCfg, c.conf.Sources, sources, event) {
		c.log.Debugf("Skipping event as it is related to recommendation informers and doesn't have any recommendations: %#v", event)
		return
	}
```

I think that explains better what's happening. Thoughts?

Reply:
Thinking out aloud (to confirm):

... The function ensures the event should be skipped if the recommendation event has no recommendation?
 
You have a recommendation "hit" and a recommendation "miss". All I can offer is: `recommendation.Miss(...)`.

The idea is not to have to dig into the guts of the function to figure out why we are returning.

The Log definitely helps. Not a blocker to shipping this.

Reply:
>... The function ensures the event should be skipped if the recommendation event has no recommendation?

Yep, that's very nice explanation what's happening here 👍 It would be great to highlight that a given function returns bool and also, as you wrote, it's about events 🤔 It can be something with `Miss`, but I find it a bit too short (it doesn't give me the same idea as `ShouldIgnoreEvent`: `Should` <- returns true/false, `Event` <- it's about events 🤔 `recommendation.Miss` could be understood in different way - `Miss` is when there's no recommendation for a given event 🤔

Reply:
Cool - makes sense. Leave as is.